packages/identity/src/verification.ts

```diff
@@ -1,14 +1,28 @@
 import { getAlgorithmParameters, getCrypto } from 'pkijs'
+import { keyObjectFromString } from '.'
 import config from './config'
+import logger from './logger'
+
+const log = logger()
 
 export const verifySignature = async (
-  userPubKey: CryptoKey,
   signature: ArrayBuffer,
-  message: string
+  message: string,
+  publicKeyObject?: CryptoKey,
+  publicKeyString?: string
 ): Promise<boolean> => {
+  if (!publicKeyObject && !publicKeyString) {
```

Comment:
this is a bit ugly...
optional positional arguments are weird, because if you wan to pass 5th without passing 4th you need to pass `undefined` explicitely. 

I suggest sth like:
```
export async verifySignature (
  userPubKey: CryptoKey,
  signature: ArrayBuffer,
  message: string
  message: string,
  publicKey: CryptoKey | string,
): Promise<boolean> {

...

if (typeOf publicKey === 'string') {
  publicKey = await keyObjectFromString(publicKey, crypto)
}
```

Reply:
also, bare in mind that reconstructing CryptoKey is expensive, so if it's done to check *every single message* it's going to cost us. It may be good to not support passing `publicKey` as string, because than user needs to hold reference to CryptoKey

Reply:
So that's what I was trying to implement in the first place. But later on you have crypto.verify method which accepts only CryptoKey